src/main/java/net/sf/jabref/logic/autosaveandbackup/BackupManager.java

```diff
@@ -0,0 +1,159 @@
+package net.sf.jabref.logic.autosaveandbackup;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import net.sf.jabref.logic.exporter.BibtexDatabaseWriter;
+import net.sf.jabref.logic.exporter.FileSaveSession;
+import net.sf.jabref.logic.exporter.SaveException;
+import net.sf.jabref.logic.exporter.SavePreferences;
+import net.sf.jabref.model.database.BibDatabaseContext;
+import net.sf.jabref.model.database.event.BibDatabaseContextChangedEvent;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import com.google.common.eventbus.Subscribe;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Backups the given bib database file from {@link BibDatabaseContext} on every {@link BibDatabaseContextChangedEvent}.
+ * An intelligent {@link ExecutorService} with a {@link BlockingQueue} prevents a high load while making backups and rejects all redundant backup tasks.
+ */
+public class BackupManager {
```

Comment:
Could you please also add a comment, that this backup is not about the bak-file which is also created upon saving.
Maybe we should distinguishes between Backup (*bak) and Live Backup (swp), or whatever we want to call these concepts.